starter-code/scripts/article.js

```diff
@@ -47,16 +47,30 @@ Article.loadAll = function(rawData) {
 // and process it, then hand off control to the View.
 Article.fetchAll = function() {
   if (localStorage.rawData) {
+    let jsonData = JSON.parse(localStorage.rawData);
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
-    //TODO: What method do we call to render the index page?
+    Article.loadAll(jsonData); //DONE: What do we pass in to loadAll()?
+    // Completed in 2 minutes - Created variable for local storage
+
+    //DONE: What method do we call to render the index page?
+    // Completed in 2 minutes - calls function from article.js
+    articleView.initIndexPage();
   } else {
-    // TODO: When we don't already have the rawData,
+    // DONE: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
     // and then render the index page.
+    //Completed in 30 minutes - Added in JSON methods for localStorage
+    const path = 'data/hackerIpsum.json';
+    $.getJSON(path).then(function(data) {
+      localStorage.setItem('rawData', JSON.stringify(data));
+      Article.fetchAll();
```

Comment:
Why are you calling Article.fetchAll inside Article.fetchAll? If we look at the if statement above, what are we calling there? Should this also be called here? Hint: yes :)